pkg/cmd/deadman/operator_condition_challenger.go

```diff
@@ -0,0 +1,195 @@
+package deadman
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configeversionedclient "github.com/openshift/client-go/config/clientset/versioned"
+	configv1versionedtypes "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions"
+	configv1lister "github.com/openshift/client-go/config/listers/config/v1"
+	clusteroperatorhelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/events"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const (
+	challengePrefix = "Requesting that the operator update this condition. "
+)
+
+type OperatorConditionChallenger struct {
+	// operatorToTimeToCheckForStaleness is a map from operator name to the time at which the condition messages will be
+	// reset to a "checking for activity" message.
+	operatorToTimeToCheckForStaleness map[string]time.Time
```

Comment:
You will need to synchronize access to this map (and the similar one in OperatorStalenessChecker) if you intend to have more than one worker.